apps/src/code-studio/components/libraries/LibraryCreationDialog.jsx

```diff
@@ -125,66 +146,94 @@ class LibraryCreationDialog extends React.Component {
   };
 
   displayFunctions = () => {
-    if (!this.state.loadingFinished) {
+    if (this.state.loadingState === LoadingState.LOADING) {
       return (
         <div style={styles.centerSpinner}>
           <Spinner />
         </div>
       );
+    } else if (this.state.loadingState === LoadingState.DONE_LOADING) {
+      let keyIndex = 0;
+      return (
+        <div>
+          <Heading2>
+            <b>{i18n.libraryName()}</b>
+            {this.state.libraryName}
+          </Heading2>
+          <form
+            ref={formElements => {
+              this.formElements = formElements;
+            }}
+            onSubmit={this.publish}
+          >
+            <textarea
+              required
+              name="description"
+              rows="2"
+              cols="200"
+              style={styles.textarea}
+              placeholder="Write a description of your library"
+            />
+            {this.state.sourceFunctionList.map(sourceFunction => {
+              let name = sourceFunction.functionName;
+              let comment = sourceFunction.comment;
+              return (
+                <div key={keyIndex} style={styles.functionItem}>
+                  <input
+                    type="checkbox"
+                    style={styles.largerCheckbox}
+                    disabled={comment.length === 0}
+                    onClick={this.validateInput}
+                    value={keyIndex++}
+                  />
+                  {name}
+                  <br />
+                  {comment.length === 0 && (
+                    <p style={styles.alert}>
+                      {i18n.libraryExportNoCommentError()}
+                    </p>
+                  )}
+                  <pre>{comment}</pre>
+                </div>
+              );
+            })}
+            <input
+              className="btn btn-primary"
+              type="submit"
+              value={i18n.publish()}
+              disabled={!this.state.canPublish}
+            />
+          </form>
+        </div>
+      );
+    } else {
+      return (
+        <div>
+          <Heading2>
+            <b>{i18n.libraryPublishTitle()}</b>
+            {this.state.libraryName}
+          </Heading2>
+          <div>
+            <p style={{fontSize: 20}}>{i18n.libraryPublishExplanation()}</p>
+            <div style={{display: 'flex', justifyContent: 'center'}}>
+              <input
+                type="text"
+                id="library-sharing"
+                onClick={select}
+                readOnly="true"
+                value={dashboard.project.getCurrentId()}
```

Comment:
NIT: Rather than getting the id from dashboard.project, it might be better to get it from the libraryClient response. That way, you know for sure you have the right project ID.

Reply:
Is there a case where the dashboard.project.getCurrentId isn't going to be the right proejct id? We should add an ticket to track that bug.

Reply:
Yeah, that's true.